pontoon/sync/tasks.py

```diff
@@ -246,6 +234,20 @@ def sync_translations(
         )
         obsolete_vcs_entities = project_changes['obsolete_db']
 
+    # If project repositories have API access, we can retrieve latest commit hashes and detect
+    # changed locales before the expensive VCS pull/clone operations. When performing full scan and
+    # when source strings have changed, we still need to sync all locales.
```

Comment:
Yeah, when I woke up I remembered I needed to add this check and while I was driving to gym I realized I remembered wrong. Will fix.